pkg/util/sqlkiller/sqlkiller_test.go

```
@@ -0,0 +1,98 @@
+		assertChanOpen(t, killer.GetKillEventChan())
+	})
+
+	t.Run("kill signal after reset clear", func(t *testing.T) {
+		killer := &SQLKiller{}
+		killSent := make(chan struct{})
+		const reason = "memory usage exceeds the instance limit"
+		testfailpoint.EnableCall(t, "github.com/pingcap/tidb/pkg/util/sqlkiller/afterResetKillSignalSwap", func() {
+			assertStateLockHeld(t, killer)
+			go func() {
+				defer close(killSent)
+				killer.SendKillSignalWithKillEventReason(KilledByMemArbitrator, reason)
```

golangci-lint (2.12.2)

[warning] line-length-limit: line is 87 characters, out of limit 80

(revive)